community/modules/scripts/ramble-setup/templates/ramble_setup.yml.tpl

```diff
@@ -12,63 +12,66 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-- name: Install Ramble
+- name: Install Application
   hosts: localhost
   vars:
+    app_name: ${app_name}
+    profile_script: ${profile_script}
     install_dir: ${install_dir}
-    ramble_url: ${ramble_url}
-    ramble_ref: ${ramble_ref}
+    git_url: ${git_url}
+    git_ref: ${git_ref}
     chmod_mode: ${chmod_mode}
     chown_owner: ${chown_owner}
     chgrp_group: ${chgrp_group}
+    finalize_setup_script: ${finalize_setup_script}
   tasks:
+  - name: Print Application Name
+    ansible.builtin.debug:
+      msg: "Running installation for application: {{app_name}}"
+
+  - name: Add profile script for application
+    ansible.builtin.copy:
+      dest: /etc/profile.d/{{ app_name }}.sh
+      mode: '0644'
+      content: "{{ profile_script }}"
+
   - name: Create parent of install directory
     ansible.builtin.file:
       path: "{{ install_dir | dirname }}"
       state: directory
 
   - name: Acquire lock
     ansible.builtin.command:
-      mkdir "{{ install_dir | dirname }}/.ramble_lock"
+      mkdir "{{ install_dir | dirname }}/.install_{{ app_name }}_lock"
     register: lock_out
     changed_when: lock_out.rc == 0
     failed_when: false
 
-  - name: Clones ramble into installation directory
-    ansible.builtin.git:
-      repo: "{{ ramble_url }}"
-      dest: "{{ install_dir }}"
-      version: "{{ ramble_ref }}"
+  - name: Clones into installation directory
+    ansible.builtin.command: git clone --branch {{ git_ref }} {{ git_url }} {{ install_dir }}
     when: lock_out.rc == 0
 
-  - name: chgrp ramble installation
+  - name: chgrp on installation
     ansible.builtin.file:
       path: "{{ install_dir }}"
       group: "{{ chgrp_group }}"
       recurse: true
     when: chgrp_group != "" and lock_out.rc == 0
 
-  - name: chown ramble installation
+  - name: chown on installation
     ansible.builtin.file:
       path: "{{ install_dir }}"
       owner: "{{ chown_owner }}"
       recurse: true
     when: chown_owner != "" and lock_out.rc == 0
 
-  - name: chmod ramble installation
+  - name: chmod on installation
     ansible.builtin.file:
       path: "{{ install_dir }}"
       mode: "{{ chmod_mode }}"
       recurse: true
     when: chmod_mode != "" and lock_out.rc == 0
 
-  - name: Check if ramble profile exists
-    ansible.builtin.stat:
-      path: /etc/profile.d/ramble.sh
-    register: profile_check
-
-  - name: Add ramble to profile
-    ansible.builtin.copy:
-      dest: /etc/profile.d/ramble.sh
-      content: ". {{ install_dir }}/share/ramble/setup-env.sh"
-    when: not profile_check.stat.exists
+  - name: Finalize Setup
+    ansible.builtin.shell: "{{ finalize_setup_script }}"
+    when: lock_out.rc == 0
```

Comment:
Could change when to have `and finalize_setup_script != ''` or something.

Reply:
Done.